scripts/pylib/twister/twisterlib.py

```diff
@@ -529,6 +530,10 @@ def _final_handle_actions(self, harness, handler_time):
         if harness_class_name == "Test":  # only for ZTest tests
             self._verify_ztest_suite_name(harness.state, harness.detected_suite_names, handler_time)
 
+            if not harness.matched_run_id and harness.run_id_exists:
+                self.set_state("failed", handler_time)
```

Comment:
this only sets the testsuite as failed, for the subcases, they will still be judged as passed.